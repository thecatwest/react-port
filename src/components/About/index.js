import React from "react";
import coverImage from "../../assets/cover/hero-portfolio.jpg";

function About() {
//   const style = {
//     display: "inline",
//     justifyContent: "center",
//     alignItems: "center",
// };
  return (
    <section className="my-5" id="about">
      <img
        src={coverImage}
        className="my-1"
        style={{ width: "100%" }}
        alt="cover"
      />
      <h1>About Me</h1>

      <p className="about-me">Hi there! I'm <span style={{ color: 'var(--secondary' }}>Cat Westover</span>, a full-stack developer, and this is my site, <a href="http://www.catwest.codes"><span style={{ color: 'var(--secondary' }}>catwest.codes.</span></a>
      <br />
      I live in <span style={{ color: 'var(--secondary' }}>Tempe, AZ</span>, where we suffer through 9 months of <span style={{ color: 'var(--secondary' }}>hot, hot, heat</span> (good band, by the way),
      <br />
      and enjoy the other 3 months to the fullest potential. When I'm not coding, I'm probably <span style={{ color: 'var(--secondary' }}>knitting socks</span> ðŸ§¶ðŸ§¦
      <br />
      
      </p>
    </section>
  );
}

export default About;
