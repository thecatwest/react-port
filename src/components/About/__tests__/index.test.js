// import react
import React from 'react';
// retrieve fns from reaact testing lib
// render function will "render" the component. What actually happens is that Jest creates a simulated DOM in a Node.js environment to approximate the DOM
// cleanup function is used to remove components from the DOM to prevent memory leaking, as well as variable or data collisions between tests that could corrupt test results.
import { render, cleanup } from '@testing-library/react';
// import the extend-expect library from the jest-dom package
import '@testing-library/jest-dom/extend-expect';
// import the component About to test
import About from '..';

// call the cleanup function using the afterEach global function from Jest
afterEach(cleanup);

// use the describe function to declare the component we're testing
describe('About component', () => {
    // First Test will be baseline to verify component is rendering
    it('renders', () => {
      render(<About />);
    });
  
    // Second Test, also known as a "test case", we'll compare snapshot versions of the DOM node structure. A snapshot is a serialized version of the DOM node structure, enabled by Jest
    it('matches snapshot DOM node structure', () => {
      // render About
      const { asFragment } = render(<About />);
      // test and compare whether the expected and actual outcomes match. Use the expect function with a matcher to assert something about a value. In the following statement, use the toMatchSnapshot matcher to assert that snapshots will match
      expect(asFragment()).toMatchSnapshot();
  });
})