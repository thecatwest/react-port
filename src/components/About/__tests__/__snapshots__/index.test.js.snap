// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section
    className="my-5"
  >
    <h1
      id="about"
    >
      about Cat
    </h1>
    <img
      alt="cover"
      className="my-2"
      src="cover-image.jpg"
      style="width: 100%;"
    />
  </section>
</DocumentFragment>
`;
